iio/iio_app/iio_demo_app.c

```diff
@@ -0,0 +1,331 @@
+/******************************************************************************/
+
+#include <stdlib.h>
+#include "error.h"
+#include "xml.h"
+#include "util.h"
+#include "iio_demo_app.h"
+#include "iio.h"
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+
+static uint32_t demo_par;
+
+static const char demo_dev_name[] = "demo_device";
+
+static const char demo_xml[] =
+	"<device id=\"demo_device\" name=\"demo_device\" >"
+	"<channel id=\"voltage0\" type=\"input\" >"
+	"<scan-element index=\"0\" format=\"le:S12/16&gt;&gt;0\" />"
+	"<attribute name=\"demo_attr\" filename=\"in_demo_attr\" />"
+	"</channel>"
+	"<channel id=\"voltage1\" type=\"input\" >"
+	"<scan-element index=\"1\" format=\"le:S12/16&gt;&gt;0\" />"
+	"</channel>"
+	"<channel id=\"voltage2\" type=\"input\" >"
+	"<scan-element index=\"2\" format=\"le:S12/16&gt;&gt;0\" />"
+	"</channel>"
+	"<channel id=\"voltage3\" type=\"input\" >"
+	"<scan-element index=\"3\" format=\"le:S12/16&gt;&gt;0\" />"
+	"</channel>"
+	"<channel id=\"voltage4\" type=\"input\" >"
+	"<scan-element index=\"4\" format=\"le:S12/16&gt;&gt;0\" />"
+	"</channel>"
+	"<channel id=\"voltage5\" type=\"input\" >"
+	"<scan-element index=\"5\" format=\"le:S12/16&gt;&gt;0\" />"
+	"</channel>"
+	"<channel id=\"voltage6\" type=\"input\" >"
+	"<scan-element index=\"6\" format=\"le:S12/16&gt;&gt;0\" />"
+	"</channel>"
+	"<channel id=\"voltage7\" type=\"input\" >"
+	"<scan-element index=\"7\" format=\"le:S12/16&gt;&gt;0\" />"
+	"</channel>"
+	"<attribute name=\"demo_attr\" filename=\"in_demo_attr\" />"
+	"</device>"
+	;
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * @brief get_rf_port_select().
+ * @param device- Physical instance of a iio_demo_device.
+ * @param buf - Where value is stored.
+ * @param len - Maximum length of value to be stored in buf.
+ * @param channel - Channel properties.
+ * @return Length of chars written in buf, or negative value on failure.
+ */
+static ssize_t get_demo_attr(void *device, char *buf, size_t len,
+			     const struct iio_ch_info *channel)
+{
+	return snprintf(buf, len, "%"PRIu32"", demo_par);
+}
+
+/**
+ * @brief set_rf_port_select().
+ * @param device - Physical instance of a iio_demo_device.
+ * @param buf - Value to be written to attribute.
+ * @param len -	Length of the data in "buf".
+ * @param channel - Channel properties.
+ * @return: Number of bytes written to device, or negative value on failure.
+ */
+static ssize_t set_demo_attr(void *device, char *buf, size_t len,
+			     const struct iio_ch_info *channel)
+{
+	demo_par = srt_to_uint32(buf);
+
+	return len;
```

Comment:
Even though is a demo app, the check + return for negative values should be implemented, since it's part of the function defined structure (as stated in the header @return: Number of bytes written to device, or negative value on failure. ).